client/my-sites/earn/memberships/index.jsx

```diff
@@ -643,117 +637,83 @@ class MembershipsSection extends Component {
+		navigateToLaunchpad();
+		fetchNextSubscriberPage( true );
+	}, [ fetchNextSubscriberPage, navigateToLaunchpad ] );
 
-		return (
-			<div>
-				<QueryMembershipsEarnings siteId={ this.props.siteId } />
-				<QueryMembershipsSettings siteId={ this.props.siteId } source={ this.props.source } />
-				{ this.props.connectedAccountId && this.renderStripeConnected() }
-				{ this.props.connectUrl && this.renderConnectStripe() }
-
-				{ ! this.props.connectedAccountId && ! this.props.connectUrl && (
-					<div className="earn__payments-loading">
-						<LoadingEllipsis />
-					</div>
-				) }
-			</div>
+	if ( ! userCan( 'manage_options', site ) ) {
+		return renderOnboarding(
+			<Notice
+				status="is-warning"
+				text={ translate( 'Only site administrators can edit Payments settings.' ) }
+				showDismiss={ false }
+			/>
 		);
 	}
+
+	return (
+		<div>
+			<QueryMembershipsEarnings siteId={ site?.ID } />
+			<QueryMembershipsSettings siteId={ site?.ID } source={ source } />
+			{ connectedAccountId && renderStripeConnected() }
+			{ connectUrl && renderConnectStripe() }
+
+			{ ! connectedAccountId && ! connectUrl && (
+				<div className="earn__payments-loading">
+					<LoadingEllipsis />
+				</div>
+			) }
+		</div>
+	);
 }
 
 /**
  * Source is used to add data to the Stripe Connect URL. On a successful
  * connection, this source is used to redirect the user the appropriate place.
  */
-const getSource = () => {
+function getSource() {
 	if ( window.location.hash === ADD_NEWSLETTER_PAYMENT_PLAN_HASH ) {
 		return 'earn-newsletter';
 	}
 	if ( window.location.hash === LAUNCHPAD_HASH ) {
 		return 'full-launchpad';
 	}
 	return 'calypso';
-};
-
-const mapStateToProps = ( state ) => {
-	const site = getSelectedSite( state );
-	const siteId = getSelectedSiteId( state );
-	const earnings = getEarningsWithDefaultsForSiteId( state, siteId );
-	const source = getSource();
+}
 
-	return {
-		site,
-		siteId,
-		siteSlug: getSelectedSiteSlug( state ),
-		total: earnings.total,
-		lastMonth: earnings.last_month,
-		forecast: earnings.forecast,
-		currency: earnings.currency,
-		commission: earnings.commission,
-		totalSubscribers: getTotalSubscribersForSiteId( state, siteId ),
-		subscribers: getOwnershipsForSiteId( state, siteId ),
-		connectedAccountId: getConnectedAccountIdForSiteId( state, siteId ),
-		connectedAccountDescription: getConnectedAccountDescriptionForSiteId( state, siteId ),
-		connectUrl: getConnectUrlForSiteId( state, siteId ),
-		hasStripeFeature:
-			siteHasFeature( state, siteId, FEATURE_PREMIUM_CONTENT_CONTAINER ) ||
-			siteHasFeature( state, siteId, FEATURE_DONATIONS ) ||
-			siteHasFeature( state, siteId, FEATURE_RECURRING_PAYMENTS ),
-		isJetpack: isJetpackSite( state, siteId ),
-		source,
-	};
-};
-
-export default connect( mapStateToProps, {
-	recordTracksEvent,
-	requestSubscribers,
-	requestDisconnectSiteStripeAccount,
-	requestSubscriptionStop,
-} )( localize( withLocalizedMoment( MembershipsSection ) ) );
```

Comment:
One small thing that isn't matched 1-for-1 in this PR is withLocalizedMoment. I played with this a bit, but it was throwing errors. So I'm just importing moment directly and using that. Using moment directly is the more common use in the calypso code base - there are 100+ files that import and use moment directly. But if we think it's a big deal to keep this, I can continue to work on that piece.

Reply:
[There's also `useLocalizedMoment`](https://github.com/Automattic/wp-calypso/blob/4413cf0ea22b368ba9f5d2da12b6ed17bb025bba/client/components/localized-moment/README.md#uselocalizedmoment) — let's use that instead.